pkg/minishift/addon/manager/addon_manager.go

```diff
@@ -266,12 +280,38 @@ func (m *AddOnManager) verifyRequiredVariablesInContext(context *command.Executi
 
 	if len(missingVars) > 0 {
 		missing := strings.TrimSpace(strings.Join(missingVars, ", "))
-		return errors.New(fmt.Sprintf("The variable(s) %s are required by the add-on, but are not defined in the context.", missing))
+		return fmt.Errorf("The variable(s) %s are required by the add-on, but are not defined in the context", missing)
 	}
 
 	return nil
 }
 
+func (m *AddOnManager) verifyRequiredOpenshiftVersion(context *command.ExecutionContext, meta addon.AddOnMeta) error {
+	dockerCommander := context.GetDockerCommander()
+	versionInfo, err := dockerCommander.Exec(" ", constants.OpenshiftContainerName, "openshift", "version")
+	if err != nil {
+		return err
+	}
+	openShiftVersionAlongWithCommitSha := strings.Split(strings.Split(versionInfo, "\n")[0], " ")[1]
+	openShiftVersion := strings.TrimSpace(strings.TrimPrefix(strings.Split(openShiftVersionAlongWithCommitSha, "+")[0], minikubeConstants.VersionPrefix))
+
+	requiredOpenshiftVersions := strings.TrimSpace(meta.OpenShiftVersion())
+	if requiredOpenshiftVersions != "" {
+		versionErr := fmt.Errorf("\nAdd-on only support listed OpenShift version semantics https://docs.openshift.org/latest/minishift/using/addons.html#addon-openshift-version-semantics "+
```

Comment:
I don't think adding URLs like this is a good idea. This becomes easily out of data.

Reply:
why do you create the error here, even though you might not need it?